index.js

```diff
@@ -338,6 +338,19 @@ class DHT extends EventEmitter {
       qs._concurrency = self.inflightQueries === 1 ? self.concurrency : backgroundCon
     }
   }
+
+  joinDht (cb) {
+    if (this.ephemeral === false) return
+    this._io._updateId(this.id)
+    this.bootstrap((err) => {
+      if (err) {
+        if (cb) cb(err)
+        return
+      }
+      this.ephemeral = false
+      if (cb) cb()
+    })
+  }
```

Comment:
Is it same to call this twice in the same tick still since we set the flag async?

Reply:
address both: 

1. not any more because we don't bail early due to func now being a bool setter
2. implemented - but blocked on testing the `setEphemeral(true)`, can't find a way to clear the cached node